src/ipz_parser.cpp

```diff
@@ -631,4 +631,31 @@ types::DbusVariantType IpzVpdParser::readKeywordFromHardware(
     return types::DbusVariantType{
         getKeywordValueFromRecord(l_record, l_keyword, l_recordOffset)};
 }
+
+void IpzVpdParser::updateRecordECC(const auto& i_recordDataOffset,
+                                   const auto& i_recordDataLength,
+                                   const auto& i_recordECCOffset,
+                                   size_t i_recordECCLength)
+{
```

Comment:
Whenever these kind of independent function is pushed, at-least make sure that it is being called from somewhere or it is just a prototype.
In this case, it is difficult to figure out if this API is actually giving the desired output or not and you can't add test output of this API to the commit message as well as it is not being called from anywhere.
The approach to break in small commits should be bottom up not the other way round.
Means, the API which needs this should be pushed first with just a comment at the place where this function is supposed to be called or call to an empty prototype of this function. Once that is in, this should be tested and pushed.